### DISCUSSION TITLE
Suggestion: remove unnecessary classes for slot variants

### DISCUSSION THREAD

**Discussion by peterbartels** [chakra-ui/panda#2476] [General]
When I use a slot recipe with a certain variant this variant is also added to slots where it's not defined. This leads to exra unnecessary classes in the class attribute

let's say I have two slots `root` and `body` where only `root` has an adjusted variant style

So a `card__body--variant_blue` is added even when there is no variant defined for this slot.

For example:
` variants:{
    variant: {
      blue: {
        root: {
          background: 'lightblue'
        }
      }
    }
  }`

I get this non existing class for body slot:
`<div class="card__body card__body--variant_blue">`

Example in playground: https://play.panda-css.com/qWJ0iRgeXK

**Comment by segunadebayo** [comment]
This sounds an interesting idea.

Looking deeply the following scenarios:
- What if a compound variants were defined that modified the “body” part
- Say this recipe was shipped as a preset, and the consumer extends the preset to include style modifications for the “body” part

I wonder how those would be managed. I’d love an approach that doesn’t lead to more runtime code.
